Joystick: Fix dig and place buttons

m42uko opened the pull request:
They were mapped to the left and right mouse buttons, which doesn't seem to work anymore in current releases. This commit changes them to report the DIG and PLACE KeyTypes.

This PR is Ready for Review.

## How to test

Try to play with a gamepad. For me, pressing the dig and place buttons did nothing. With this patch, they work as expected.

SmallJoker commented:
Please also update `game.cpp` starting with `input->joystick.clearWasKeyDown(KeyType::MOUSE_L);`
It is likely that those KeyType values can be removed from `keys.h`. Would be amazing if you could test that change as well. Thanks.

m42uko commented:
@SmallJoker Thanks. Changed my first commit to also clear the events and added a second commit to remove the KeyTypes from `keys.h`. They are not used anywhere in the code anymore.

m42uko commented:
Hmm looking at the code right now, I see `SCROLL_UP` and `SCROLL_DOWN` are also exclusive to joystick input, but we seem to have keyboard equivalents as `HOTBAR_PREV` and `HOTBAR_PREV`. The `SCROLL_*` KeyTypes still work, but I guess it would be a good idea to change them to prevent the same thing that happened to `MOUSE_L` and `MOUSE_R`.

What do you think? Is there anything special about the `SCROLL` events when compared to `HOTBAR`?

m42uko commented:
I combined everything into a single joystick cleanup commit now. If the thing with the `SCROLL` KeyTypes is a bad idea, or if I should still split it into multiple commits, let me know and I'll do so.

sfan5 reviewed (approved):
Non-joystick functionality unaffected, code looks good.

SmallJoker reviewed (approved):
LGTM
